wordpress-mobile/WordPress-Android
# Issue/13268 on item click

Partially fixes #13268

This PR adds onItemClick listener to items on the Activity Type Filter screen. The DiffUtil is added so the item selector's animation doesn't get interrupted.

Merge instructions:
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/13469 is merged
2. Remove "Not ready for merge" label
3. Merge this PR


To test:
- Make sure the ActivityLogFiltersFeatureConfig is enabled
1. My site -> Activity Log
2. Click on "Activity Type" chip
3. Click on an unchecked row and notice it gets checked
4. Click on a checked row again and notice it gets unchecked

PR submission checklist:
